taichi/transforms/scalarize.cpp

```diff
@@ -47,7 +47,12 @@ class Scalarize : public BasicStmtVisitor {
       auto dest_tensor_type = dest_dtype->template as<TensorType>();
       auto val_tensor_type = val_dtype->template as<TensorType>();
 
-      TI_ASSERT(dest_tensor_type->get_shape() == val_tensor_type->get_shape());
+      // Taichi used to treat VectorType(n) as MatrixType(n, 1),
```

Comment:
Probably worth discussing how we represent Vectors in CHI IR. Would be better to make the semantics consistent across different levels of IRs.

Reply:
+1 for further discussion over Vector v.s. 1-dimensional Matrix semantics.

The other thing is that `type_check` probably does not allow `Vector(n) op Matrix(n, 1)`, but `scalarization` pass stays synchronized with the legacy `scalarization` happened in Python, which is why we decided to let `Vector(n) op Matrix(n, 1)` pass.

Let's have a thorough discussion then decide what we should do with this PR.

Reply:
SGTM :)